UnboundLib/Cards/CustomCard.cs

```diff
@@ -213,16 +221,13 @@ private static void ResetOnlyGunStats(Gun gun)
             gun.isProjectileGun = false;
             gun.defaultCooldown = 1f;
             gun.attackSpeedMultiplier = 1f;
-            gun.objectsToSpawn = new ObjectsToSpawn[0];
+            gun.objectsToSpawn = Array.Empty<ObjectsToSpawn>();
             gun.projectileColor = Color.black;
         }
 
         private static void ResetOnlyCharacterStatModifiersStats(CharacterStatModifiers characterStatModifiers)
         {
-            for (int i = 0; i < characterStatModifiers.objectsAddedToPlayer.Count; i++)
-            {
-                GameObject.Destroy(characterStatModifiers.objectsAddedToPlayer[i]);
-            }
+            foreach (var attachedObject in characterStatModifiers.objectsAddedToPlayer) Destroy(attachedObject);
```

Comment:
@willis81808 I agree it looks like it could, but `Destroy` doesn't actually destroy the object when it's called. It just marks the object to be destroyed whenever unity is ready. `DestroyImmediate` would cause issues though. Either way, it's probably best to change this just in case.

Reply:
Most recent change is functionally identical to how it was before, and potentially subject to the same error. The original was iterating over the collection in reverse order, but technically if we're not also removing the objects from this collection in the same loop we should be fine.

Reply:
I just reverted the change I made